tests/phpunit/unit-tests/models/class-llms-test-model-llms-lesson.php

```diff
@@ -417,4 +417,46 @@ public function test_is_orphan() {
 
 	}
 
+	/**
+	 * Test the new functionality for the course level setting for prerequiesites
+	 */
+	public function test_course_level_prerequisite()
```

Comment:
this particular test is (mostly) okay. It would be better to move this test to the course tests class.

And then add tests for the two lesson methods that have been affected by the pr.


Despite what a lot of our existing tests put forth as a "style" we've learned that making smaller more "atomic" tests is better and more maintainable over time.

Therefore the lesson tests class should have a few new tests to test the various of each method. EG: don't write one test to test all the functions in one test function, instead write multiple test functions to test each potential condition of each new function.

For example `has_prerequisite()` could have multilpe tests:

test_has_prerequisite_course_sequential -- tests the return when the course has sequential enabled
test_has_prerequisite_on -- no course prereq but does have lesson enabled
test_has_prereq_off -- no course && no lesson

Reply:
Done & Added new test cases